aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -203,5 +204,61 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool,
+      "The output tensor of lt must be a bool, but was ", result.scalar_type());
+  auto iter = TensorIterator::comparison_op(result, self, other,
+      /*check_mem_overlap=*/true);
+  lt_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor lt(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  native::lt_out(result, self, other);
+  return result;
+}
+
+Tensor& lt_(Tensor& self, const Tensor& other) {
+  TORCH_CHECK(self.dtype() == other.dtype(),
+      "Expected object of scalar type ", self.dtype(), " but got scalar type ",
+      other.dtype(), " for argument 'other' in call to lt_")
+  auto iter = TensorIterator::comparison_op(self, self, other,
+      /*check_mem_overlap=*/true);
+  lt_stub(iter.device_type(), iter);
+  return self;
+}
+
+// First it will validate that is possible to convert scalar to tensor dtype without overflow.
+// It's different for arithmetic ops. For arithmetic ops we don't verify this.
+// The reason is because previous TH behavior checked for overflow in comparison ops but not arithmetic ops.
+// In the future, we should reconsider this inconsistency and decide if we're going to check overflow for arithmetic ops
+Tensor& lt_out(Tensor& result, const Tensor& self, Scalar other) {
+  AT_DISPATCH_ALL_TYPES_AND3(at::ScalarType::Bool, at::ScalarType::BFloat16, at::ScalarType::Half, self.scalar_type(), "lt_out", [&]{
+    // Validate that is possible to convert scalar to tensor dtype without overflow
+    native::lt_out(result, self, wrapped_scalar_tensor(other.to<scalar_t>()));
+  });
+  return result;
+}
+
+Tensor lt(const Tensor& self, Scalar other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  AT_DISPATCH_ALL_TYPES_AND3(at::ScalarType::Bool, at::ScalarType::BFloat16, at::ScalarType::Half, self.scalar_type(), "lt", [&]{
+    // Validate that is possible to convert scalar to tensor dtype without overflow
+    native::lt_out(result, self, wrapped_scalar_tensor(other.to<scalar_t>()));
+  });
+  return result;
+}
+
+Tensor& lt_(Tensor& self, Scalar other) {
+  AT_DISPATCH_ALL_TYPES_AND3(at::ScalarType::Bool, at::ScalarType::BFloat16, at::ScalarType::Half, self.scalar_type(), "lt_", [&]{
+    // Validate that is possible to convert scalar to tensor dtype without overflow
+    auto iter = TensorIterator::comparison_op(self, self, wrapped_scalar_tensor(other.to<scalar_t>()),
```

Comment:
We're "validating that it is possible to convert scalar to tensor dtype` in multiple places. Could we turn this into a helper function? i.e.,
`static Tensor wrapped_scalar_tensor_as(Scalar, Tensor)`

Reply:
Fixed